File: testdata/bench/go/BenchmarkTest00107.go
package bench

import (
	"fmt"
	"net/http"
)

// XSS via fmt.Fprint
func Handler00107(w http.ResponseWriter, r *http.Request) {
	title := r.URL.Query().Get("title")
	w.Header().Set("Content-Type", "text/html")
	fmt.Fprint(w, "<title>"+title+"</title>")
}

File: testdata/bench/kotlin/BenchmarkTest00107.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*

// Vulnerable: open redirect via header value
suspend fun handler00107(call: ApplicationCall) {
    val referer = call.request.headers["Referer"]
    call.respondRedirect(referer ?: "/")
}
